src/new_api_v04/constraint.rs

```diff
@@ -28,6 +28,30 @@ pub enum LiteralEvalError {
     UnboundVariable(String),
 }
 
+/// Errors that occur when constructing constraints.
+#[derive(Clone, Debug, Error)]
+pub enum InvalidConstraint {
```

Comment:
Add some context to the errors; e.g. print the relevant predicate and constraint.

Reply:
Thanks good idea, I have added better descriptions.